region/maridia/inner-pink.json

```diff
@@ -156,15 +156,28 @@
                       "fromNode": 1,
                       "framesRemaining": 60,
                       "shinesparkFrames": 40
-                    }}
-                  ]
+                    }},
+                    {"or": [
+                      "canCrouchJump",
+                      "Gravity",
+                      "HiJump",
+                      "canSpringBallJumpMidAir"
+                    ]}
+                  ],
+                  "devNote": "This does not require canWaterShineCharge, as you can store the shinecharge before entering the room."
```

Comment:
Oasis has the same situation.  Many places probably do.
`canWaterShineCharge` may need to be rolled into `canComeInCharged`.  
There is charging a spark before the room.  Charging the spark in room with gravity.  And Charging the spark in room with `canWaterShineCharge`.  And its a pain to separate those cases here.

It looks like Gravity+Shinespark without canSuitlessMaridia is not a working combination across strats too.